addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MavenMetadataGenerator.java

```diff
@@ -158,6 +158,8 @@
 
     private final ReentrantLock mergerLock = new ReentrantLock();
 
+    private static final int THREAD_WAITING_TIME_SECONDS = 5;
```

Comment:
Let's set this to something like 300 seconds. That way if it takes a long time it'll still complete...but it won't stay deadlocked forever.

Does that sound reasonable?

Reply:
Will change it.